waiter/src/waiter/metrics.clj

```diff
@@ -473,3 +474,10 @@
    :stream-read-body (service-timer service-id "stream-read-body")
    :stream-request-rate (service-meter service-id "stream-request-rate")
    :throughput-meter (service-meter service-id "stream-throughput")})
+
+(defn report-duration
+  "Report elapsed duration on the given timer metric. Returns nil."
+  [timer start-time end-time]
```

Comment:
Please add the type hint on `timer` here instead of the `.update` call.

Reply:
Done.